Description: Workarounds for dpkg-shlibdeps warnings
 This patch gives SONAMEs to liblibcodelite and libplugin. This fixes the
 dpkg-shlibdeps warnings and allows codelite-plugins to "properly" link against
 libraries from the codelite package.
 .
 Converting the other libraries to static libs is ok since they are only used
 by one library. It also reduces the final size and fixes some more warnings.
Author: James Cowgill <jcowgill@debian.org>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CodeLite/CMakeLists.txt
+++ b/CodeLite/CMakeLists.txt
@@ -112,6 +112,7 @@
 
 # Define the output
 add_library(libcodelite SHARED ${SRCS})
+set_target_properties(libcodelite PROPERTIES SOVERSION 0)
 if (UNIX AND NOT APPLE )
     target_link_libraries(libcodelite 
                          ${LINKER_OPTIONS} 
--- a/Plugin/CMakeLists.txt
+++ b/Plugin/CMakeLists.txt
@@ -89,6 +89,7 @@
 
 # Define the output
 add_library(plugin SHARED ${SRCS})
+set_target_properties(plugin PROPERTIES SOVERSION 0)
 if(GTK2_FOUND)
     target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${GTK2_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite)
 elseif (GTK3_FOUND)
--- a/sdk/databaselayer/CMakeLists.txt
+++ b/sdk/databaselayer/CMakeLists.txt
@@ -71,7 +71,7 @@
 endif ( WITH_MYSQL )
 
 # Define the output
-add_library(databaselayersqlite SHARED ${SRCS} ${MYSQL_SRCS})
+add_library(databaselayersqlite STATIC ${SRCS} ${MYSQL_SRCS})
 target_link_libraries(databaselayersqlite ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES})
 
 if(APPLE)
--- a/sdk/wxshapeframework/CMakeLists.txt
+++ b/sdk/wxshapeframework/CMakeLists.txt
@@ -36,7 +36,7 @@
 FILE(GLOB SRCS "src/*.cpp" "src/wxxmlserializer/*.cpp")
 
 # Define the output
-add_library(wxshapeframework SHARED ${SRCS})
+add_library(wxshapeframework STATIC ${SRCS})
 target_link_libraries(wxshapeframework ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES})
 
 if (NOT MINGW)
